### DISCUSSION TITLE
Autodiscovery floppy format

### DISCUSSION THREAD

**Discussion by flyman02** [keirf/greaseweazle#349] [Ideas]
HI,
Congratulations, the project is increasingly complete.
Who knows if the GW software will ever have a self -recognition function of the floppy format.
There is a program called "HxCFloppyEmulator" which in the floppy dump function can find the format independently.
I saw that there is also the source code code, maybe it can help for ideas.
Thanks and continue for this way
Fly

**Comment by keirf** [comment]
If you use --format=ibm.scan then Greaseweazle will autodetect any IBM based track format. That's most apart from Apple GCR, Amiga and C64.

**Reply by flyman02** [reply]
Thanks, I have several floppies without description and I think a function that returns only disk information and structure data would be very useful. It would be a nice upgrade. Thanks and good work

**Comment by qbarnes** [comment]
As an alternative, you can also use `gw2dmk`.  It does autorecognition of most all FM and MFM formatted floppies of any size.  It handles mixed format disks (even mixing FM and MFM sectors within the same track).  It does the autorecognition of the disk's format and reading all its tracks in a single pass.  The utility is liked by archivists who get stacks of floppies in unknown formats.  You typically just need to tell it what drive type and media size you're reading from with the `-k` option, pop the disk in, and let `gw2dmk` go.

The utility writes only DMK save files.  DMK is a fairly low-level floppy save file (usually able to handle most common floppy copy-protection too), but unlike raw formats, `gw2dmk` also detects CRC errors and will automatically retry (reread).  With `gw` now supporting DMK save files, you can use it to convert a DMK to many other formats as needed.

gw2dmk has prebuilt binaries for Windows 32-/64-bit and Linux (ARM and x86_64) (https://github.com/qbarnes/gw2dmk/releases).  If you need it for MacOS, let me know and I'll work with you to get it ported to that platform.  BTW, I'm overly cautious and call its current release beta, but it works solidly for most all formats and drives just fine.

If you have any questions about how to use it or how it works, just ask!
